src/report/report.service.ts

```diff
@@ -14,19 +14,19 @@ export class ReportService {
         private readonly userService: UserService,
     ) {}
 
-    getAllReports(): Promise<Report[]> {
+    find(): Promise<Report[]> {
         return this.model.find().exec();
     }
 
     findById(id: string): Promise<Report> {
         return this.model.findById(id).exec();
     }
 
-    getUserReports(): Promise<Report[]> {
+    findUserReports(): Promise<Report[]> {
         return this.model.find({ type: ReportType.User }).exec();
     }
 
-    getSystemReports(): Promise<Report[]> {
+    findSystemReports(): Promise<Report[]> {
         return this.model.find({ type: ReportType.System }).exec();
```

Comment:
You can remove duplicated `find()`, `findSystemReports()` and `findUserSystems()` by accept a `filters` in `find` like this:

```ts
find(filters): Promise<Report[]> {
    return this.model.find(filters).exec();
}
```
And then just pass these filters from controller.